packages/api/lib/writeRecords/write-execution.js

```diff
@@ -79,53 +80,53 @@ const buildExecutionRecord = ({
   });
 };
 
-const writeExecutionViaTransaction = async ({
+const _writeExecutionRecord = ({
+  dynamoRecord,
+  postgresRecord,
+  knex,
+  executionModel = new Execution(),
+  executionPgModel = new ExecutionPgModel(),
+}) => knex.transaction(async (trx) => {
+  logger.info(`About to write execution ${postgresRecord.arn} to PostgreSQL`);
+  const [executionCumulusId] = await executionPgModel.upsert(trx, postgresRecord);
+  logger.info(`Successfully wrote execution ${postgresRecord.arn} to PostgreSQL with cumulus_id ${executionCumulusId}`);
+
+  await executionModel.storeExecutionRecord(dynamoRecord);
+  return executionCumulusId;
+});
+
+const writeExecution = ({
   cumulusMessage,
+  knex,
   collectionCumulusId,
   asyncOperationCumulusId,
   parentExecutionCumulusId,
-  trx,
-  executionPgModel = new ExecutionPgModel(),
-  updatedAt,
+  executionModel = new Execution(),
+  updatedAt = Date.now(),
 }) => {
-  const executionRecord = buildExecutionRecord({
+  const postgresRecord = buildExecutionRecord({
     cumulusMessage,
     collectionCumulusId,
     asyncOperationCumulusId,
     parentExecutionCumulusId,
     updatedAt,
   });
-  logger.info(`About to write execution ${executionRecord.arn} to PostgreSQL`);
-  const upsertResponse = await executionPgModel.upsert(trx, executionRecord);
-  logger.info(`Successfully wrote execution ${executionRecord.arn} to PostgreSQL with cumulus_id ${upsertResponse[0]}`);
-  return upsertResponse;
+  const dynamoRecord = Execution.generateRecord(cumulusMessage, updatedAt);
+  return _writeExecutionRecord({ dynamoRecord, postgresRecord, knex, executionModel });
 };
 
-const writeExecution = async ({
-  cumulusMessage,
+const writeExecutionApiRecord = async ({
```

Comment:
I have no issues with the refactors, but can we rename this function to `writeExecutionRecordsFromApi` and the one above to `writeExecutionRecordsFromMessage`? I had to read the code a couple times to understand the changes and these names might make it more obvious